### ISSUE TITLE
IgnoreAllNonExisting support in 5.0

### ISSUE THREAD

**Issue by crowbarsolutions** [AutoMapper/AutoMapper#1386]
In the last few releases I've been using Mapping extensions like this:

		public static IMappingExpression IgnoreAllNonExisting(this IMappingExpression expression)
		{
			foreach (var property in expression.TypeMap.GetUnmappedPropertyNames())
			{
				expression.ForMember(property, opt => opt.Ignore());
			}
			return expression;
		}

		public static IMappingExpression IgnoreAllNonExistingSource(this IMappingExpression expression)
		{
			foreach (var property in expression.TypeMap.GetUnmappedPropertyNames())
			{
				expression.ForSourceMember(property, opt => opt.Ignore());
			}
			return expression;
		}

Which I use in my configuration like so:

		CreateMap(typeof(TempMapViewModel), typeof(MapViewModel))
				.IgnoreAllNonExisting()
				.IgnoreAllNonExistingSource()
				.ForAllMembers(opt => opt.Condition(srs => !srs.IsSourceValueNull));

This currently doesn't work because `expression.TypeMap` is no longer available. Is there a better alternative in 5.0?

**Comment by jbogard** [comment]
What are you trying to do with all those ignores?

**Comment by crowbarsolutions** [comment]
I have properties in MapViewModel that you could consider required, and I have nullable properties in TempMapViewModel that are optional. I want to take all the properties properties in TempMapViewModel that are not null and merge them into MapViewModel.

**Comment by jbogard** [comment]
No I meant why are you ignoring all those members?

**Comment by crowbarsolutions** [comment]
By default, AutoMapper tries to map all properties of the source type to the destination type. If some of the properties are not available in the destination type it will not throw an exception when doing the mapping.

I override this behavior with the extension method to ignore all properties that do not exist on the target type.

**Comment by jbogard** [comment]
No, it doesn't. It only throws an exception during configuration validation, never at runtime for missing members.

I think you need `CreateMap<Source, Dest>(MemberList.Source)`, this uses the source type as validation.

**Comment by crowbarsolutions** [comment]
Yes that's what I meant, no exception during runtime, but an exception during config validation.

Your suggestion is precisely what I was looking for, thanks.

**Comment by crowbarsolutions** [comment]
Now that 5.0.1 had come out I had to change this to `CreateMap<Source, Dest>(MemberList.None)` otherwise I would get the error: 

> No available constructor.

**Comment by webpaul** [comment]
MemberList.None appears to have no affect on behavior for me in 6.0.2 I had the following 4.2 code that would map a single property from an object that has hundreds:

var map = CreateMap<Source, Dest>();
map.ForAllMembers(x => x.Ignore());
map.ForMember(x => x.Prop1, o => o.MapFrom(x => x.Prop1))

The "IgnoreAll" now has the behavior of ignoring even though I have set a mapping for it after ignoring. I tried adding MemberList.None to the CreateMap call but it is throwing an error on properties I don't care about. Suggestions?
